crates/matrix-sdk/src/room/common.rs

```diff
@@ -205,16 +205,17 @@ impl Common {
         #[cfg(feature = "e2e-encryption")]
         if let Some(machine) = self.client.olm_machine().await {
             for event in http_response.chunk {
-                let decrypted_event =
-                    if let AnySyncRoomEvent::MessageLike(AnySyncMessageLikeEvent::RoomEncrypted(
-                        SyncMessageLikeEvent::Original(encrypted_event),
-                    )) = event.deserialize_as::<AnySyncRoomEvent>()?
-                    {
-                        machine.decrypt_room_event(&encrypted_event, room_id).await?
-                    } else {
-                        RoomEvent { event, encryption_info: None }
-                    };
+                if let Ok(AnySyncRoomEvent::MessageLike(AnySyncMessageLikeEvent::RoomEncrypted(
+                    SyncMessageLikeEvent::Original(encrypted_event),
+                ))) = event.deserialize_as::<AnySyncRoomEvent>()
+                {
+                    if let Ok(event) = machine.decrypt_room_event(&encrypted_event, room_id).await {
+                        response.chunk.push(event);
+                        continue;
```

Comment:
I'm not a big fan of this extra control flow. Can we keep the `let decrypted_event =` and just have two `else { RoomEvent { event, encryption_info: None }` branches until we can refactor this pattern to if-let-chains everywhere?

Reply:
sure, i thought you preferred this way :)